qcengine/programs/orca.py

```diff
@@ -0,0 +1,297 @@
+    _defaults: Dict[str, Any] = {
+        "name": "Orca",
+        "scratch": True,
+        "thread_safe": False,
+        "thread_parallel": True,
+        "node_parallel": False,
+        "managed_memory": True,
+    }
+    version_cache: Dict[str, str] = {}
+
+    # Set of implemented dft functionals in Orca according to the manual.
+    # fmt: off
+    _dft_functionals: Set[str] = {
+        "HFS" ,"LDA" , "LSD" , "VWN", "VWN5" , "VWN3" , "PWLDA" , "BP86" ,
+        "BLYP" , "OLYP" , "GLYP" , "XLYP" , "PW91" , "mPWPW" , "mPWLYP" ,
+        "PBE" , "RPBE" , "REVPBE" , "PWP" , "B1LYP" , "B3LYP" , "B3LYP/G" ,
+        "O3LYP" , "X3LYP" , "B1P" , "B3P" , "B3PW" , "PW1PW" , "PBE0" ,
+        "PW6B95" , "BHANDHLYP" , "TPSS" , "TPSS0" , "M06L" , "M062X" , "wB97"
+        , "wB97X" , "wB97X-D3" , "CAM-B3LYP" , "LC-BLYP" , "B2PLYP" ,
+        "RI-B2PLYP" , "B2PLYP" , "B2PLYP-D" , "B2PLYP-D3" , "RI-B2PLYP" ,
+        "mPW2PLYP-D" , "B2GP-PLYP" , "B2K-PLYP" , "B2T-PLYP" , "PWPB95" ,
+        "RI-PWPB95" , "D3BJ" , "D3ZERO" , "D2"
+        }
+
+    # fmt: on
+
+    # NOTE: Unrestricted SCF methods must be specified by using keyword reference
+    _hf_methods: Set[str] = {"HF", "RHF"}
+    _restricted_post_hf_methods: Set[str] = {"MP2", "CCSD", "CCSD(T)"}  # RMP2, RCCSD, RCCSD(T)}
+    # TODO Add keyword to specify unrestricted for WF method
+    # _unrestricted_post_hf_methods: Set[str] = {"UMP2", "UCCSD", "UCCSD(T)"}
+    _post_hf_methods: Set[str] = {*_restricted_post_hf_methods}
+
+    class Config(ProgramHarness.Config):
+        pass
+
+    def found(self, raise_error: bool = False) -> bool:
+        return which(
+            "orca",
+            return_bool=True,
+            raise_error=raise_error,
+            raise_msg="Please install via https://orcaforum.kofo.mpg.de/app.php/portal.",
+        )
+
+    # TODO Consider changing this to use molpro --version instead of performing a full execute
+    def get_version(self) -> str:
+        self.found(raise_error=True)
+
+        # name_space = {"molpro_uri": "http://www.molpro.net/schema/molpro-output"}
+        # which_prog = which("molpro")
+        # if which_prog not in self.version_cache:
+        #     success, output = execute(
+        #         [which_prog, "version.inp", "-d", ".", "-W", "."],
+        #         infiles={"version.inp": ""},
+        #         outfiles=["version.out", "version.xml"],
+        #     )
+
+        #     if success:
+        #         tree = ET.ElementTree(ET.fromstring(output["outfiles"]["version.xml"]))
+        #         root = tree.getroot()
+        #         version_tree = root.find("molpro_uri:job/molpro_uri:platform/molpro_uri:version", name_space)
+        #         year = version_tree.attrib["major"]
+        #         minor = version_tree.attrib["minor"]
+        #         molpro_version = year + "." + minor
+        #         self.version_cache[which_prog] = safe_version(molpro_version)
+
+        return "4.2.1"
```

Comment:
if orca doesn't do a `orca --version` or similar, often programs will print a version in their output file. try experimenting with a very minimal input that we can sneak the version out of the output header.

Reply:
Yeah. I realized you do something like it for Molpro if I am not wrong. Let me come up with a hacky solution for it.